.github/workflows/cmake.yml

```diff
@@ -60,6 +60,7 @@ jobs:
         -DSEQUANT_SKIP_LONG_TESTS=${{ matrix.valgrind || matrix.sanitize }}
         -DSEQUANT_COMPILE_DOC_EXAMPLES=${{ !matrix.valgrind }}
         -DSEQUANT_PYTHON=${{ !matrix.valgrind && !matrix.sanitize }}
+        -DSEQUANT_ASSERT_BEHAVIOR=THROW
```

Comment:
I don't think I like having `throw` as the default here. This opens up the possibility of absorbing assertion failures (unintentionally) in `try…exept` blocks (and the message printed to console won't cause CI to fail)`.

I would propose overwriting the macro definition only for the macros test file and leaving it at `abort` for everything else.